project-canvas/src/components/StoryMapView/CaseSubActionLevel.tsx

```diff
@@ -0,0 +1,49 @@
+import { Group, Stack } from "@mantine/core"
+import { StrictModeDroppable } from "../common/StrictModeDroppable"
+import { SubActionGroup } from "./SubActionGroup"
+import { Case, SubAction } from "./types"
+
+export function CaseSubActionLevel({
+  filteredCases,
+  levelId,
+  addSubAction,
+  updateSubAction,
+}: {
+  filteredCases: Case[]
+  levelId: string
+  addSubAction: (actionId: string, subAction: SubAction) => void
+  updateSubAction: ({ id, title }: SubAction) => void
+}) {
+  return (
+    <Group align="start">
+      {filteredCases.map((caseColumn) => (
+        <Group key={`${caseColumn.id}-${levelId}`} align="start" spacing={0}>
+          {caseColumn.actions
+            .map((_action) => _action.subActionGroups)
+            .flat()
+            .map(({ id: subActionGroupId, subActions }) => (
+              <SubActionGroup
+                key={subActionGroupId}
+                subActionGroupId={subActionGroupId}
+                subActions={subActions}
+                addSubAction={addSubAction}
+                updateSubAction={updateSubAction}
+              />
+            ))}
+
+          <StrictModeDroppable key="add" droppableId="add" isDropDisabled>
+            {(provided) => (
+              <Stack
+                w="162px"
```

Comment:
(just flagging this) might lead to mistakes if other values are changed in the future

Reply:
Will look into removing that in the next PR